README.md

```diff
@@ -46,7 +69,77 @@ npm install
 
 ## Usage
 
-TBD
+This client assumes:
+
+* a remote Wallet Attached Storage server is available
+* the app is able to create and store a public-private key pair, see the
+  [App Identity and Key Management](#app-identity-and-key-management) section
+  for more details.
+
+Example key pair generation (for an in-browser SPA), for persistence in local
+storage.
+
+```ts
+import { Ed25519Signer } from '@did.coop/did-key-ed25519'
+
+const appDidSigner = await Ed25519Signer.generate()
+```
+
+Example storing and loading:
+
+```ts
+// Serialize to JSON for storage
+const exportedKeyPair = appDidSigner.toJSON()
+localStorage.setItem('app-DID', JSON.stringify(exportedKeyPair))
+
+// Load from storage and turn back into a DID Signer
+const loadedKeyPair = localStorage.getItem('app-DID')
+const appDidSigner = Ed25519Signer.fromJSON(loadedKeyPair)
+```
+
+Create a Wallet Attached Storage Client:
+
+```ts
+import { WasClient } from '@did-coop/wallet-attached-storage'
+
+const config = {
+  was: {
+    url: 'https://data.pub'
+  }
+}
+
+const storage = new WasClient({ url: config.was.url, signer: appDidSigner })
+```
+
+Provision (create and set up) a Data Space (this performs a [Create Space HTTP
+API operation](https://did-coop.github.io/wallet-attached-storage-spec/#create-space-operation)):
+
+```ts
+let space
+try {
+  space = await storage.space('demo-space')
+} catch (e) {
+  console.error('Error initializing data space:', e)
+  throw e
+}
+```
+
+Now you can read and write resources to and from collections:
+
+```ts
+// Create a request to list the contents of the 'credentials' collection
+const cursor = space.collection('credentials').list()
+
+// Iterate through the results
+for await (const vc of cursor) {
+  console.log(vc);
+}
+// or cast the results to an array
+const allItems = await cursor.toArray()
+
+// Write a VC to a collection
+const { url } = await space.collection('credentials').put(myVc)
```

Comment:
This is confusingly different from @data.pub/fetch-client, where this would PUT a representation of the collection... not append an item to a collection...